### ISSUE TITLE
bring stackoverflow questions into the course - set precendent in week 1

### ISSUE THREAD

**Issue by jsms90** [foundersandcoders/master-reference#301]
Great point from @otajor 
![image](https://cloud.githubusercontent.com/assets/15656538/26681437/15a494c0-46d5-11e7-8e7e-055c031571cf.png)

It's been said before by @nelsonic @sohilpandya and probably many others. Let's do it!

We would need this in week 1 (could even be mentioned at Monday's installation party?):
+ [ ] Make sure everyone has a Stack Overflow account
+ [ ] Mention in the week 1 project intro

Longer term perhaps
+ [ ] Tips for where/how to start & links to good examples

What do you think week 1 mentors?

**Comment by bradreeder** [comment]
@jsms90 I'm unsure if we should be advising students to raise questions as a requirement when my understanding was that best practice on stack overflow would be to first identify if your problem has already been addressed in a different thread? I haven't used it in a long time so maybe I'm mistaken.

All for encouraging students to make use of it though. 👍

**Comment by eliascodes** [comment]
I'm echoing @bradreeder here.

If this is about awareness of SO as an avenue, then cool, good idea. Otherwise though, I'd be broadly conscious of the SO community's tendency to be relatively impatient with people who ask for stuff that can be easily found already. They're unlikely to be asking stuff in the first part of the course that really requires a new SO question.

**Comment by jsms90** [comment]
@bradreeder Good point. You probably don't want to mandate a question a week. I would say an interaction on SO per week.

**Comment by edgarfinn** [comment]
I believe @Samatar26 is planning on making one of the research topics for week1 into researching techniques. Personally I think students should be familiarised with:

+ Googling techniques, and refactoring search queries when initial searches don't yield sufficient answers
+ Stackoverflow best practises ([https://stackoverflow.com/help/how-to-ask](https://stackoverflow.com/help/how-to-ask) is a great resource.
+ Technology documentation, and quickly getting what you need out of it. 

I'm sure there's more, but I'm happy to work with Samatar on the research topic guide. 

And I agree with Elias and Bradley. I don't think its a big priority to get people up-voting and asking questions on SO. Just getting them thinking critically about how to learn, and comfortable with the resources available will be a HUGE step forward IMO

**Comment by Samatar26** [comment]
I don't think I have ever come across a problem which hasn't been answered before while on the course. I think it's more helpful towards the end or maybe after the course, but making them aware of and how to use SO is definitely a great thing, but not necessarily making them ask a question each week which has probably been answered before. Including a research topic right from the start of the course including SO will definitely help the students tremendously.

**Comment by jsms90** [comment]
To be clear, I think the greatest benefit to this is to help students join that community. If students can be part, not just of the FAC community, but an online community that is as large as Stack Overflow, that would be really really useful.

I'm suggesting that we should help students into whilst on the course, _because_ it's notoriously harsh on beginners.

But I totally agree with everything that's been said above. Given @eliascodes' point, maybe we should think about this for week 11?
> They're unlikely to be asking stuff in the first part of the course that really requires a new SO question.

The 3 week product build and move towards self-directed learning might be the perfect time?

**Comment by Samatar26** [comment]
I agree that they should be introduced to SO in week 1 and it will be part of the research topics
